rcl/include/rcl/wait.h

```diff
@@ -0,0 +1,306 @@
+// Copyright 2015 Open Source Robotics Foundation, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+#ifndef RCL__WAIT_H_
+#define RCL__WAIT_H_
+
+#if __cplusplus
+extern "C"
+{
+#endif
+
+#include <stdbool.h>
+#include <stddef.h>
+
+#include "rcl/subscription.h"
+#include "rcl/guard_condition.h"
+#include "rcl/time.h"
+#include "rcl/types.h"
+
+struct rcl_wait_set_impl_t;
+
+/// Container for subscription's, guard condition's, etc to be waited on.
+typedef struct rcl_wait_set_t {
```

Comment:
Maybe, though I think it would be passed as an option to the `rcl_wait_set_init` function and the result would be hidden behind an accessor and stored in the `impl` pointer.